api/ngrok/v1alpha1/cloudendpoint_types.go

```diff
@@ -0,0 +1,92 @@
+/*
+MIT License
+
+Copyright (c) 2024 ngrok, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudEndpointSpec defines the desired state of CloudEndpoint
+type CloudEndpointSpec struct {
+	// The unique URL for this cloud endpoint. This URL is the public address
+	// +kubebuilder:validation:Required
+	URL string `json:"url"`
+
+	// TrafficPolicyRef is a reference to the TrafficPolicy resource to attach to the Cloud Endpoint
+	// +kubebuilder:validation:Optional
+	TrafficPolicyName string `json:"trafficPolicyName,omitempty"`
+
+	// TrafficPolicy allows inline definition of a TrafficPolicy object
+	// +kubebuilder:validation:Optional
+	TrafficPolicy *NgrokTrafficPolicySpec `json:"trafficPolicy,omitempty"`
```

Comment:
What do you think about nesting these 2 under a single `trafficPolicy`? Not sure on the API, but thinking something similar to what k8s does for env vars. Maybe something like?

```
spec:
  trafficPolicy:
    value: "raw traffic policy here"
```

or

```
spec:
  trafficPolicy:
    name: "
```

Reply:
hmmm, they are pretty similar. I kind of feel like these 2 separate fields feel more separate than if they are nested it might be more confusing thinking that you can add both a name and a value, but I guess you could fix that by calling it like a ref or something. 

I'm inclined to keep it as it is unless you prefer it this way

Reply:
Nope, no strong preference. Just a curiosity.